akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -438,19 +436,22 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
       log.debug("Started embedded media driver in directory [{}]", driver.aeronDirectoryName)
       topLevelFREvents.loFreq(Transport_MediaDriverStarted, driver.aeronDirectoryName().getBytes("US-ASCII"))
       Runtime.getRuntime.addShutdownHook(stopMediaDriverShutdownHook)
-      mediaDriver = Some(driver)
+      mediaDriver.set(Some(driver))
```

Comment:
make it a CAS and fail if it was set before. Unlikely but better fail than having two drivers concurrently or something even more weirder.